lib/views/pay/pay_view.dart

```diff
@@ -1,102 +1,214 @@
 import 'package:clnapp/components/buttons.dart';
+import 'package:clnapp/model/app_model/decode_invoice.dart';
 import 'package:clnapp/model/app_model/pay_invoice.dart';
 import 'package:clnapp/utils/app_provider.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
+import 'numberpad_view.dart';
 
 class PayView extends StatefulWidget {
   final AppProvider provider;
 
   const PayView({Key? key, required this.provider}) : super(key: key);
+
   @override
   State<PayView> createState() => _PayViewState();
 }
 
 class _PayViewState extends State<PayView> {
-  String state = "";
   String? boltString;
-  int? amountMsat;
-  AppPayInvoice? paymentResponse;
-  String? error;
+  final _invoiceController = TextEditingController();
+  late String display;
+
+  Future<AppPayInvoice> payInvoice(String boltString) async {
+    final response =
+        await widget.provider.get<AppApi>().payInvoice(invoice: boltString);
+    return response;
+  }
 
-  Future<AppPayInvoice> payInvoice(String boltString, int? amountMsat) async {
-    // we don't need error handling here
-    final response = await widget.provider
-        .get<AppApi>()
-        .payInvoice(invoice: boltString, msat: amountMsat);
-    setState(() {
-      state = "Payment Status ${response.payResponse.status}";
-    });
+  Future<AppDecodeInvoice> decodeInvoice(String invoice) async {
+    final response = await widget.provider.get<AppApi>().decodeInvoice(invoice);
     return response;
   }
 
-  Widget _buildMainView() {
+  Widget _buildMainView(BuildContext context) {
+    var size = MediaQuery.of(context).size;
     return Padding(
-      padding: const EdgeInsets.all(30.0),
+      padding: EdgeInsets.symmetric(horizontal: size.width * 0.08),
       child: Column(
         mainAxisAlignment: MainAxisAlignment.spaceEvenly,
         children: [
           TextField(
+            minLines: 5,
+            maxLines: 10,
+            controller: _invoiceController,
             onChanged: (invoiceBolt) {
               boltString = invoiceBolt;
             },
-            decoration: const InputDecoration(
-              border: OutlineInputBorder(),
+            decoration: InputDecoration(
+              enabledBorder: OutlineInputBorder(
+                borderRadius: BorderRadius.circular(20),
+                borderSide: const BorderSide(
+                  color: Color.fromARGB(255, 255, 121, 197),
+                  width: 1.0,
+                ),
+              ),
+              focusedBorder: OutlineInputBorder(
+                borderRadius: BorderRadius.circular(20),
+                borderSide: const BorderSide(
+                  color: Color.fromARGB(255, 255, 121, 197),
+                  width: 1.0,
+                ),
+              ),
+              alignLabelWithHint: true,
+              contentPadding: const EdgeInsets.all(20),
               labelText: 'Invoice Bolt11/12',
               hintText: 'Bolt11/12',
```

Comment:
```suggestion
              hintText: 'Invoice',
```

Bolt11 is an invoice, so we keed just the invoice here and in all the places that you use `BOlt11/12`